src/NCalc/Domain/EvaluationVisitor.cs

```diff
@@ -74,7 +74,14 @@ public int CompareUsingMostPreciseType(object a, object b)
         {
             var allowNull = (_options & EvaluateOptions.AllowNullParameter) == EvaluateOptions.AllowNullParameter;
 
-            Type mpt = allowNull ? GetMostPreciseType(a?.GetType(), b?.GetType()) ?? typeof(object) : GetMostPreciseType(a.GetType(), b.GetType());
+            if (!allowNull)
+            {
+                if (a == null)
+                    throw new ArgumentNullException(nameof(a));
+
+                if (b == null)
+                    throw new ArgumentNullException(nameof(b));

```

Comment:
```
ArgumentNullException.ThrowIfNull(a, nameof(a));
ArgumentNullException.ThrowIfNull(b, nameof(b));
```

Reply:
This is not supported by netstandard 2.0 and .net framework 4.5